src/api/environment.cc

```diff
@@ -529,6 +529,9 @@ NODE_EXTERN std::unique_ptr<InspectorParentHandle> GetInspectorParentHandle(
   CHECK_NOT_NULL(env);
   if (name == nullptr) name = "";
   CHECK_NE(thread_id.id, static_cast<uint64_t>(-1));
+  if (!env->should_create_inspector()) {
+    return nullptr;
+  }
```

Comment:
I’m not familiar with C++, can you explain what this does and how it fixes the issue?

Reply:
the issue was when creating a worker - it used the same inspector server as the parent/main thread.
but the assumption that there is such a server was wrong in some cases (e.g the parent process of `--test`)

Reply:
I assumed `should_create_inspector` explains the part of ignoring this code path when we don't have an inspector, and the content of  `should_create_inspector` explains when that happens (`--test`, `--watch` etc)